src/bio/bio_device.c

```diff
@@ -468,11 +460,46 @@ bio_replace_dev(struct bio_xs_context *xs_ctxt, uuid_t old_dev_id,
 }
 
 static int
-json_write_cb(void *cb_ctx, const void *data, size_t size)
+json_write_bdev_cb(void *cb_ctx, const void *json, size_t json_size)
 {
-	struct bio_dev_info	*b_info = cb_ctx;
-	char			*prefix = "traddr\": \"";
-	char			*traddr, *end;
+	struct bio_dev_info *b_info = cb_ctx;
+
+	return bio_decode_bdev_params(b_info, json, (int)json_size);
+}
+
+static int
+json_find_bdev_params(struct spdk_bdev *bdev, struct bio_dev_info *b_info)
+{
+	struct spdk_json_write_ctx *json;
+	int                         rc;
+
+	json = spdk_json_write_begin(json_write_bdev_cb, b_info, SPDK_JSON_WRITE_FLAG_FORMATTED);
+	if (json == NULL) {
+		D_ERROR("Failed to alloc SPDK json context\n");
+		return -DER_NOMEM;
+	}
+
+	rc = spdk_bdev_dump_info_json(bdev, json);
+	if (rc != 0) {
```

Comment:
[Question] Do we need even vall spdk_json_write_end() or some other API to free spdk json context memory here?

Reply:
yes, I have added to  both usages, fixed